### PR TITLE
New OR-Set implementation, should save muchos bytos

### PR DESCRIPTION
**Pull request by lenary**
Github got confused and closed #49, so work is here (we were screwing around with branches)

Needed: 
- [ ] Benchmark.

### CODE REVIEW

**Comment by lenary** [comment]
I've rebased onto `develop` and added stuff about precondition errors and context. The model still passes.

I haven't run dialyzer. It probably hates me.

File: src/riak_dt_orset.erl [hunk 1]
```diff
@@ -42,7 +42,7 @@
 -endif.
 
 -export_type([orset/0, binary_orset/0, orset_op/0]).
--opaque orset() :: {entries(), entries()}.
+-opaque orset() :: orddict:orddict(member(), uniques()).
```
Comment (seancribbs) [inline src/riak_dt_orset.erl]: `orddict` is a 0-arity type:

```
orddict() = [{Key :: term(), Value :: term()}]
```

File: src/riak_dt_orset.erl [hunk 2]
```diff
@@ -189,8 +122,13 @@ reset(ORSet, Actor) ->
 %% an operation is needed at a replica without sending the entire
 %% state to the client.
 -spec precondition_context(orset()) -> orset().
-precondition_context({ADict, _RDict}) ->
-    {ADict, orddict:new()}.
+precondition_context(ORDict) ->
+    ordict:fold(fun(Elem, Tokens, ORDict1) ->
```
Comment (seancribbs) [inline src/riak_dt_orset.erl]: Typo, `ordict` -> `orddict`

**Comment by seancribbs** [comment]
I've run the `orset_bm` against this, it uses way more space than the ORSWOT (proportional to the number of adds), but is not significantly slower for it. :+1:, I'll handle the merge.

**Comment by lenary** [comment]
:hammer: :hand:

**Pull request merged**